gobblin-runtime/src/main/java/gobblin/runtime/TaskContext.java

```diff
@@ -132,6 +132,9 @@ public WriterOutputFormat getWriterOutputFormat(int branches, int index) {
   public List<Converter<?, ?, ?, ?>> getConverters(int index) {
     String converterClassKey =
         ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.CONVERTER_CLASSES_KEY, index);
+
+    this.workUnitState.setProp(ConfigurationKeys.FORK_BRANCH_ID_KEY, index);
```

Comment:
The use is so that each converter can be aware of which branch it is being called in. This allows for the feature described in the description of the pull request. So it looks like TaskContext.getConverters(int) gets called in each Fork operator, and each Fork operator shares the same TaskContext object. The TaskContext.getConverters() method is where the init method of each constructor is called, which is why I added in the line there. Where in the Fork operator can I put this code? I need access to the WorkUnitState that is used to initialize each converter. 

The Forks can run concurrently, but wouldn't they created and initialized sequentially.